File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/signus/CreateKeyTest.java
package org.hyperledger.indy.sdk.signus;

import org.bitcoinj.core.Base58;
import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.hyperledger.indy.sdk.InvalidStructureException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.*;


public class CreateKeyTest extends IndyIntegrationTestWithSingleWallet {

	@Test
	public void testCreateKeyWorksForSeed() throws Exception {
		String paramJson = new SignusJSONParameters.CreateKeyJSONParameter(MY1_SEED, null).toJson();
		String senderVk = Signus.createKey(wallet, paramJson).get();
		assertEquals(32, Base58.decode(senderVk).length);
	}

	@Test
	public void testCreateKeyWorksWithoutSeed() throws Exception {
		String senderVk = Signus.createKey(wallet, "{}").get();
		assertEquals(32, Base58.decode(senderVk).length);
	}

	@Test
	public void testCreateKeyWorksForInvalidSeed() throws Exception {
		String paramJson = new SignusJSONParameters.CreateKeyJSONParameter("invalid_base58_string11111111111", null).toJson();

		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		Signus.createKey(wallet, paramJson).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/SignusTests/CreateKeyTest.cs
using Hyperledger.Indy.SignusApi;
using Hyperledger.Indy.Test.Util.Base58Check;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.SignusTests
{
    [TestClass]
    public class CreateKeyTest : IndyIntegrationTestWithSingleWallet
    {
        [TestMethod]
        public async Task TestCreateKeyWorksForSeed()
        {
            var paramJson = string.Format("{{\"seed\":\"{0}\"}}", MY1_SEED);
            var senderVk = await Signus.CreateKeyAsync(wallet, paramJson);
            Assert.AreEqual(32, Base58CheckEncoding.DecodePlain(senderVk).Length);
        }

        [TestMethod]
        public async Task TestCreateKeyWorksWithoutSeed()
        {
            var senderVk = await Signus.CreateKeyAsync(wallet, "{}");
            Assert.AreEqual(32, Base58CheckEncoding.DecodePlain(senderVk).Length);
        }

        [TestMethod]
        public async Task TestCreateKeyWorksForInvalidSeed()
        {
            var paramJson = string.Format("{{\"seed\":\"{0}\"}}", "invalidSeedLength");

            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
               Signus.CreateKeyAsync(wallet, paramJson)
           );
        }
    }
}
